### PR TITLE
enhancement/required-account-fields

### PR DESCRIPTION
**Pull request by br0tda**
Added logic to mark as required src/dst account based on transaction type 

---
name: Pull request
about: AutocompleteField isRequired attribute 
title: Logic for required account fields
labels: feature
assignees: victorbalssa

---

**Is your feature request related to a problem? Please describe.**
Currently, the source account is always marked as required, even in a 'deposit'  transaction

**Describe the solution you'd like**
The source account is marked as required only in 'withdrawal' or 'transfer' transaction.
The destination account is marked as required only in 'deposit' or 'transfer' transaction.

**Describe alternatives you've considered**
...

**Additional context**
...

### CODE REVIEW

File: src/components/Forms/TransactionSplitForm.tsx [hunk 1]
```diff
@@ -320,7 +320,7 @@ export default function TransactionSplitForm({
       />
 
       <AutocompleteField
-        isRequired
+        isRequired={'withdrawal' === formData.type || 'transfer' === formData.type }
```
Comment (victorbalssa) [inline src/components/Forms/TransactionSplitForm.tsx]:
```suggestion
        isRequired={['withdrawal', 'transfer'].includes(formData.type)}
```

File: src/components/Forms/TransactionSplitForm.tsx [hunk 2]
```diff
@@ -340,6 +340,7 @@ export default function TransactionSplitForm({
       />
 
       <AutocompleteField
+        isRequired={'deposit' === formData.type || 'transfer' === formData.type }
```
Comment (victorbalssa) [inline src/components/Forms/TransactionSplitForm.tsx]:
```suggestion
        isRequired={['deposit', 'transfer'].includes(formData.type)}
```

**Review by victorbalssa — CHANGES_REQUESTED** [review]
Thank you @br0tda!

I like the idea, can you update the suggestions so I can deploy this in the next minor version?

**Comment by br0tda** [comment]
Hi @victorbalssa 

Code updated. Thank you!

**Review by victorbalssa — APPROVED** [review]

**Pull request merged**